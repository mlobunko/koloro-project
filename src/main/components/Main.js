import React from "react";

import Description from "app/description";
import Form from "app/form";
import Logo from "app/logo/components";

import ErrorBoundary from "./ErrorBoundary";
import {
  StyledMain,
  StyledTextBlock,
  StyledImageGameOfThrones,
  StyledGradient
} from "./styles";

export class Main extends React.Component {
  state = {
    isLandscape: true
  };
  landscapeOn = () => {
    this.setState({
      isLandscape: true
    });
  };
  landscapeOff = () => {
    this.setState({
      isLandscape: false
    });
  };
  checkLandscape = () =>
    document.documentElement.clientWidth > document.documentElement.clientHeight
      ? this.landscapeOn()
      : this.landscapeOff();

  componentDidMount = () => {
    this.checkLandscape();
    window.addEventListener("resize", this.checkLandscape);
  };
  render() {
    return (
      <ErrorBoundary>
        <StyledGradient>
          <StyledMain landscape={this.state.isLandscape}>
            <StyledTextBlock>
              <Logo landscape={this.state.isLandscape} />
              <Description landscape={this.state.isLandscape} />
              <Form landscape={this.state.isLandscape} />
            </StyledTextBlock>
            <StyledImageGameOfThrones landscape={this.state.isLandscape} />
          </StyledMain>
        </StyledGradient>
      </ErrorBoundary>
    );
  }
}

Main.displayName = "Main";

export default Main;
