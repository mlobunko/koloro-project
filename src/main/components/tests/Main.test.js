import React from "react";
import { shallow } from "enzyme";

import { Main } from "../Main";

describe("landscapeOn", () => {
  it("set isLandscape to true", () => {
    const wrapper = shallow(<Main />);
    wrapper.setState({ isLandscape: false });
    wrapper.instance().landscapeOn();
    expect(wrapper.state("isLandscape")).toEqual(true);
  });
});

describe("landscapseOff", () => {
  it("set isLandscape to false", () => {
    const wrapper = shallow(<Main />);
    wrapper.setState({ isLandscape: true });
    wrapper.instance().landscapeOff();
    expect(wrapper.state("isLandscape")).toEqual(false);
  });
});

describe("checkLandscape", () => {
  const clientHeight = 200;
  const clientWidth = 100;

  Object.defineProperty(document, "documentElement", {
    value: { clientHeight, clientWidth }
  });

  it("call landscapeOn", () => {
    const wrapper = shallow(<Main />);
    wrapper.instance().landscapeOn = jest.fn();
    wrapper.instance().landscapeOff = jest.fn();
    wrapper.instance().checkLandscape();
    expect(wrapper.instance().landscapeOff).toHaveBeenCalledTimes(1);
    expect(wrapper.instance().landscapeOn).toHaveBeenCalledTimes(0);
  });

  it("call landscapeOff", () => {
    document.documentElement.clientHeight = 100;
    document.documentElement.clientWidth = 200;
    const wrapper = shallow(<Main />);
    wrapper.instance().landscapeOn = jest.fn();
    wrapper.instance().landscapeOff = jest.fn();
    wrapper.instance().checkLandscape();
    expect(wrapper.instance().landscapeOff).toHaveBeenCalledTimes(0);
    expect(wrapper.instance().landscapeOn).toHaveBeenCalledTimes(1);
  });
});

describe("componentDidMount", () => {
  Object.defineProperty(window, "addEventListener", {
    value: jest.fn()
  });

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("call checkLandscape", () => {
    const wrapper = shallow(<Main />);
    wrapper.instance().checkLandscape = jest.fn();
    wrapper.instance().componentDidMount();
    expect(wrapper.instance().checkLandscape).toHaveBeenCalledTimes(1);
  });

  it("call addEventListener", () => {
    const wrapper = shallow(<Main />);
    expect(window.addEventListener).toHaveBeenCalledWith(
      "resize",
      wrapper.instance().checkLandscape
    );
  });
});

describe("render", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
