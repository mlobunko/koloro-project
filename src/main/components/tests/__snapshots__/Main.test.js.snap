// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<ErrorBoundary>
  <StyledGradient>
    <StyledMain
      landscape={true}
    >
      <StyledTextBlock>
        <Logo
          clientWidth={200}
          landscape={true}
        />
        <Component
          landscape={true}
        />
        <Component
          landscape={true}
        />
      </StyledTextBlock>
      <StyledImageGameOfThrones
        landscape={true}
      />
    </StyledMain>
  </StyledGradient>
</ErrorBoundary>
`;
